st2client/st2client/shell.py

```diff
@@ -78,6 +78,11 @@
     'debug': ['cli', 'debug']
 }
 
+# A list of command classes for which automatic authentication should be skipped.
+SKIP_AUTH_CLASSES = [
+    'TokenCreateCommand'
```

Comment:
First I actually check for "auth" being first value in argv, but then I decided to go with the current approach...

I'm also working on some tests, but it's a bit of a pita.